tests/Unit/HTTP/FileClientTest.php

```diff
@@ -75,4 +75,32 @@ public function testFileClientProvidesCorrectOptionsAndHeaders(): void
             ['Accept' => 'application/atom+xml']
         );
     }
+
+    public function testFileClientReturnsResponseWithStatusCode429(): void
+    {
+        $response = $this->createStub(File::class);
+        $response->method('get_status_code')->willReturn(429);
+
+        $registry = $this->createStub(Registry::class);
+        $registry->method('create')->willReturn($response);
+
+        $client = new FileClient($registry, [
+            'timeout' => 30,
+            'useragent' => 'dummy-useragent',
+            'redirects' => 10,
+            'force_fsockopen' => true,
+            'curl_options' => [
+                \CURLOPT_FAILONERROR => true,
+            ],
+        ]);
```

Comment:
What is the point of setting the options instead of defaults? Also the last two are mutually exclusive.

```suggestion
        $client = new FileClient($registry);
```